tests/BenchmarkDotNet.Tests/ArtifactFileNameHelperTests.cs

```diff
@@ -0,0 +1,87 @@
+﻿using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Diagnosers;
+using BenchmarkDotNet.Helpers;
+using BenchmarkDotNet.Running;
+using BenchmarkDotNet.Tests.XUnit;
+using System.Buffers.Tests;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace BenchmarkDotNet.Tests
+{
+    public class ArtifactFileNameHelperTests
+    {
+        [FactWindowsOnly(nonWindowsSkipReason: "ETW Sessions can be created only on Windows")]
+        public void OnWindowsWeMustAlwaysUseOldLongPathsLimitForSessionFiles()
+        {
+            var config = DefaultConfig.Instance
+                .WithArtifactsPath(@"C:\Projects\performance\artifacts\bin\MicroBenchmarks\Release\netcoreapp5.0\BenchmarkDotNet.Artifacts");
+
+            var benchmarkCase = BenchmarkConverter.TypeToBenchmarks(typeof(RentReturnArrayPoolTests<byte>), config).BenchmarksCases.First();
+
+            var parameters = new DiagnoserActionParameters(
+                process: null,
+                benchmarkCase: benchmarkCase,
+                new BenchmarkId(0, benchmarkCase));
+
+            foreach (string fileExtension in new[] { "etl", "kernel.etl", "userheap.etl" })
+            {
+                var traceFilePath = ArtifactFileNameHelper.GetTraceFilePath(parameters, new System.DateTime(2020, 10, 1), fileExtension);
+
+                Assert.InRange(actual: traceFilePath.Length, low: 0, high: 260);
+            }
+        }
+    }
+}
+
+namespace System.Buffers.Tests
+{
+    [GenericTypeArguments(typeof(byte))] // value type
+    [GenericTypeArguments(typeof(object))] // reference type
+    public class RentReturnArrayPoolTests<T>
```

Comment:
Why do you need such a complicated example?

Reply:
@WojciechNagorski to get a benchmark with a very long name. I was lazy and I just copied it 1:1 from the bug report